rfcs/0012-mtls.md

```diff
@@ -0,0 +1,106 @@
+# RFC Template
+
+- Feature Name: `mtls`
+- Start Date: 2024-11-27
+- RFC PR: [Kuadrant/architecture#110](https://github.com/Kuadrant/architecture/pull/110)
+- Issue tracking: [Kuadrant/kuadrant-operator#1049](https://github.com/Kuadrant/kuadrant-operator/issues/1049)
+
+# Summary
+[summary]: #summary
+
+MTLS support for communication between the Wasm plugin in the gateway pod, and Authorino & Limitador.
+
+# Motivation
+[motivation]: #motivation
+
+For security reasons so that all traffic is encrypted, and certificates validated.
+This aligns with a 'zero trust' architecture.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Update the Kuadrant CR to have a new configuration option `mtls` for users to configure if mtls should be enabled for either authorino or limitador.
+The reason for having separate flags for either component is to allow flexibility where the resource overhead of tls may be considered too high when enabled for a specific component e.g. for checking limits on each request in limitador.
+The new config in the Kuadrant CR would look like this:
+
+```yaml
+apiVersion: Kuadrant.io/v1alpha1
+kind: kuadrant
+spec:
+  mtls:
+    authorino: true
+    limitador: true
```

Comment:
Although this could work for a first implementation, I wonder if it's clear to the user that the certs will be provisioned via [Secret discovery service (SDS)](https://www.envoyproxy.io/docs/envoy/latest/configuration/security/secret).

I can also see how non-mesh users could want this but by configuring certs by other means though. If we don't support such other means right away, should the API be more explicit about using SDS perhaps?

Reply:
So the available options right now for mTLS seem quite limited (from reading this recent article https://blog.howardjohn.info/posts/mtls-kubernetes/)
There's either a sidecar approach (which this uses), ambient mode, a CNI approach (which doesn't exist yet), or DIY.
Is the abstraction too low if SDS is mentioned in the config?
What about an mTLS mode of 'mesh' being the initial option, with that meaning we'll integrate with the mesh if it's available i.e. if you're using Istio.

Reply:
One thing I am thinking here is we could go with an opinionated mTLS but ensure we allow users to still change values directly if needed. So when reconciling, we would assert that the section is there but not overwrite any new config for example. 
So out of the box you get a working version that uses SDS if you want more than that you can configure it directly and we wont overwrite it